### DISCUSSION TITLE
RESTful Naming Conventions for Inference APIs

### DISCUSSION THREAD

**Discussion by jiyer2016** [bentoml/BentoML#1324] [Q&A] [answered]
I'd like to name the inference API into the more RESTful looking - "/faces/detect".
But I get the following error:

`bentoml.exceptions.InvalidArgument: Invalid API name: '/faces/detect', a valid identifier may only contain letters, numbers, underscores and not starting with a number`

Is there any reason this is disallowed ?

Can this restriction be removed if possible ?

**Comment by parano** [comment, marked as answer]
The inference API name has this constraint because it is also used as the python API, CLI command, AWS resource name, and potentially K8S resource name.

We are adding an API for the user to defined additional HTTP routes for accessing their BentoService here https://github.com/bentoml/BentoML/issues/1161 with a more general API like `add_url_route`, would it help with your use case?

**Reply by jiyer2016** [reply]
I lose a lot of goodies if use the more general API. Is there a way by which this additional route can point back into the same inference api ? It doesn't hurt if the api is exposed via two routes - `/detect` and `/faces/detect`.

**Reply by parano** [reply]
@jiyer2016 we can potentially add an API for changing just the URL endpoint when serving via HTTP model server, something like `@api(http_url_route='/faces/detect')`, would that work for your use case?

**Reply by jiyer2016** [reply]
That works I guess. An alternate suggestion could be to do something like `@api(additional_http_routes=('/abc/def', '/ghi/jkl/mno'))`

That would retain the original route for backward compatibility and also add additional ones.

Having said that you original solution on `http_url_route` works perfectly fine.

**Reply by jiyer2016** [reply]
Thanks - May I know when this feature is likely to be released ? It is not critical - but just for my own planning

**Reply by parano** [reply]
@bojiang has a PR out for this already here https://github.com/bentoml/BentoML/pull/1334 - most likely it will be released in the next release within the next 2 weeks
